lib/order.rb

```diff
@@ -0,0 +1,55 @@
+require_relative 'menu'
+
+class Order
+  attr_reader :menu, :cost, :items
+  DELIVERY_TIME = 1800
+
+  def initialize(menu = Menu.new)
+    @menu = menu.list
+    @items = {}
+    cost = 0
+  end
+
+  def update(items)
+    manage_order(items)
+    calculate_cost
+  end
+
+  def confirm
+    "Thank you! Your order was placed and will be delivered before #{Time.new + DELIVERY_TIME}."
+  end
+
+  private
+
+  def manage_order(items)
+    items = items.split(", ")
+    items.each { |dish|
+    reformat_order(dish)
+    }
+  end
+
+  def reformat_order(dish)
+    dishes = dish.split(" ")
+    dish = dishes[1].to_sym
+    number = dishes[0].to_i
+    check_order(dish, number)
+    save_order(dish, number)
+  end 
+
+  def check_order(dish, number)
```

Comment:
I'd look into implementing the check_order() into save_order, OR look at calling it within save_order. This will still be following SRP, but will make the code flow better and save you typing `order.check_order` as well as `order.save_order`